src/CMakeLists.txt

```diff
@@ -11,10 +11,25 @@ set(UMF_CUDA_INCLUDE_DIR
     ""
     CACHE PATH "Directory containing the CUDA headers")
 
+# Compose the UMF_ALL_CMAKE_VARIABLES variable containing all the CMake
+# variables that contain the "UMF" string in the name and do not contain the '#'
+# character in the value. Preprocessor definitions containing '#' cannot be
+# passed on the compiler command line because many compilers do not support it.
+get_cmake_property(_variableNames VARIABLES)
+foreach(_variableName ${_variableNames})
+    if(("${_variableName}" MATCHES "UMF") AND (NOT "${${_variableName}}" MATCHES
+                                               "#"))
+        set(UMF_ALL_CMAKE_VARIABLES
+            "${_variableName}:${${_variableName}},${UMF_ALL_CMAKE_VARIABLES}")
+    endif()
+endforeach()
```

Comment:
perhaps it'd be nice to sort the list after it's composed (based on the example you pasted it looks like `UMF_VERSION_REVISION` is the first one to show up, which seems a little weird, as this var is not really useful 😉 )

Reply:
What for? If I would sort it, it could have not very useful order too. Usefulness of the order depends on a user.

Reply:
Done